### ISSUE TITLE
Enable debug logging level for all classes in Anko Logger

### ISSUE THREAD

**Issue by ajaysaini-sgvu** [Kotlin/anko#489]
Hi I am using `anko` logger in my android application. I see that I have to set like this  ` setprop  log.tag.LoginActivity DEBUG`  for my every class where I am doing logging. 

Execute this command for each new class file is not a good solution.  Is there any way to enable debug log level for entire application. I am doing logging like below

```
debug { "Registration Activity" }

```

I see that to solve this issue just create one class and do something like this 

```
public class Logger {
    static String TAG = "MyApp";
     
    public static void v(String str) {
        if (Log.isLoggable(TAG, Log.VERBOSE)) {
            Log.v(TAG, str);
        }
    }
    public static void v(String str, Throwable t) {
        if (Log.isLoggable(TAG, Log.VERBOSE)) {
            Log.v(TAG, str, t);
        }
    }
}
```

Then use like this `Logger.v("Hello World");` and enable debug level using below command for entire application.

```
setprop  log.tag.MyApp  VERBOSE
```

Does anko provide solution for this problem without making own `Logger`  class and one TAG for entire application ?

**Comment by Rawa** [comment]
I think this issue needs more attention. The ankologger is not very user-friendly in it's current state.

**Comment by gaffonso** [comment]
I agree that Anko is doing "the right thing" by checking isLoggable() behind the scenes before it calls Log.  That's an important optimization to preserve.

The fact that, by default, isLoggable() returns false for VERBOSE and DEBUG logging levels isn't terribly convenient for devs.  It is, effectively, acting as a pre-filter to the log entries shown in Logcat. (For dev builds, I'd prefer to have all log entries "written out" and then just filter once they get to Logcat.)

The general guidance for enabling VERBOSE and DEBUG in Anko is to change the isLoggable default behavior when initiating runs via ADB by setting a system property on the command line to affect this, something like `log.tag.MyAppTag=WARN`

HOWEVER

1. Affecting the Anko log-level via a log.tag system property allows you to affect only a single tag-name at a time (Anko, with its "terse" logging statments auto-sets the tag to the classname).  This means you cannot "enable debug logging" for the entire application using this technique, you need to enable it on a tag-by-tag basis.

2. It's stupid-hard to find guidance on how to change the behavior of isLoggable (the log.tag System Property) when *not* initiating runs from adb but instead initiating runs via an Android Studio run configuration.  Android studio run configurations provide no way I can see to "set system properties".

+1 for spending some more time getting this "right".  In the least I would suggest the following bare-minimum functionality:

I would suggest either:

1.  Improving the documentation for Anko Logging such that it's clear how to override the isLoggable() defaults in a variety of situations/contexts (notably, when running via Android Studio Run Configurations).

OR (better yet)

2. Change the underlying mechanism by which Anko determines what to "pre-filter".  Let whatever the underlying flag is be able to affect logging app-wide (not just a single tag) AND ensure that whatever the underlying flag is that's used to determine pre-filter behavior, that it's compatible with Android Studio Run Configurations.  Perhaps do *not* use isLoggable() checks, they seem to be an arcane little corner of Android that's poorly documented, at least WRT to Angular Studio.

**Comment by berkus** [comment]
Anything being done on this front or it's just generally not recommended to use `AnkoLogger` now?

**Comment by gmikhail** [comment]
Any news?

**Comment by Flo354** [comment]
Also interested. We can't use it for now

**Comment by Wavesonics** [comment]
Ya I wish there was an option to just skip the isLoggable check, it's such a pain other wise. Might have to switch to another logger.
